Library/Homebrew/language/python.rb

```diff
@@ -88,6 +88,15 @@ def self.setup_install_args(prefix)
       ]
     end
 
+    def self.rewrite_python_shebang(python_path)
+      Dir["**/*"].each do |f|
+        next unless File.file?(f)
+        next unless File.read(f).include?("/usr/bin/env python")
+
+        Utils::Inreplace.inreplace f, %r{#! ?/usr/bin/env python}, "#! #{python_path}"
```

Comment:
- These 2 lines check for different substrings, in theory `Utils::Inreplace.inreplace` could fail if a file has a comment with "/usr/bin/env python", for example (i.e. not a real shebang)
- I'm not sure, but it looks like `#! /usr/bin/env python2` and `#! /usr/bin/env python3` will be rewriten not as expected (i.e. `#! /usr/bin/env python2` will be replaced to `#! /usr/local/Cellar/python/3.7.6_1/bin/python32` which doesn't exist)

Reply:
> * I'm not sure, but it looks like `#! /usr/bin/env python2` and `#! /usr/bin/env python3` will be rewriten not as expected (i.e. `#! /usr/bin/env python2` will be replaced to `#! /usr/local/Cellar/python/3.7.6_1/bin/python32` which doesn't exist)

What are the specific cases we need to replace/not replace?



> * Maybe we need to handle `#! /usr/bin/python` as well?

What would be the case in which we need to do this?

Reply:
> What are the specific cases we need to replace/not replace?

I think we want to replace the following
- `#! /usr/bin/env python` (the most common one)
- `#! /usr/bin/env python2`
- `#! /usr/bin/env python2.7`
- `#! /usr/bin/env python3` (this one exists in `glib` for example)
- `#! /usr/bin/env python3.X`

> What would be the case in which we need to do this?

If a formula has`#! /usr/bin/python` then we need to put `uses_from_macos "python@2"` which we actually don't really use much (to help linuxbrew, as far as I understand it)

Reply:
I have not seen `#! /usr/bin/python` that much lately, but I may be wrong.

I want only to replace:
`#! /usr/bin/python`
`#! /usr/bin/env python`
`#! /usr/bin/env python2`
`#! /usr/bin/env python2.7`

These are fine:
`#! /usr/bin/env python3`
`#! /usr/bin/env python3.X`

Of course the python 3 ones may point to whatever python 3 is on the system. We might now want to do that and enforce our version of Python 3 everywhere. But this means basically using this rewrite function in probably more than 400 formulae: we need and audit for this and enforce it.

At the end we should get rid of `uses_from_macos "python@2"`, when it is completely gone.